caffe2/CMakeLists.txt

```diff
@@ -410,6 +415,7 @@ if(NOT INTERN_BUILD_MOBILE OR NOT BUILD_CAFFE2_MOBILE)
       $<$<BOOL:${SELECTED_OP_LIST}>:--selected-op-list-path="${SELECTED_OP_LIST}">
       --force_schema_registration
       --gen_lazy_ts_backend
+      ${GEN_PER_OPERATOR_FLAG}
```

Comment:
hmm I might just be misunderstanding. But why does the actual codegen need to know if we're building the per-operator flag or not?

I think the way it works today is that the codegen will always emit the per-operator headers. The codegen templates files themselves include an IFDEF like this:
```
#ifndef AT_PER_OPERATOR_HEADERS
#include <ATen/Operators.h>
#else
${operator_headers}
#endif
```

Example template: https://github.com/pytorch/pytorch/blob/14bf20cd922c3ba33c32343c19fd9ac490d4f7a6/aten/src/ATen/templates/RegisterFunctionalization.cpp#L9

Reply:
I think I didn't realize this logic existed, but it is still not enough since I also had to fix this (see below)

```'ops_headers': '#include <ATen/Functions.h>' if not per_operator_headers else '',```



Perhaps we could move ATen/Functions.h into the template ifdef and then things would have been OK.

Reply:
I think I didn't realize this logic existed, but it is still not enough since I also had to fix this (see below)

```'ops_headers': '#include <ATen/Functions.h>' if not per_operator_headers else '',```



Perhaps we could move ATen/Functions.h into the template ifdef and then things would have been OK.

Reply:
> I think the way it works today is that the codegen will always emit the per-operator headers. The codegen templates files themselves include an IFDEF like this:

I think this isn't quite correct, the `RegisterFunctionalization` template is a nice thing to emulate but it is not how `RegisterDispatchKey` works currently.  RegisterDispatchKey currently has its `per_operator_headers` thing hardcoded to False in gen_backend_stubs.py (probably since that is presumed only used out of tree), and RegisterDispatchKey.cpp doesn't have the nice #ifndef that you pointed out in RegisterFunctionalization.

I was starting to mess with this, and I am kind of wondering if it's ok to land as-is and then update how it works.  I mainly don't want to add any more delay to landing the TS backend since that is blocking other stuff downstream.

Reply:
ok yep, I definitely don't want to hold up the PR on this if landing this PR is actively blocking other things.